### PR TITLE
Fix broken provide rule

### PR DESCRIPTION
**Pull request by stof**
| Q               | A
| --------------- | ---
| Bug fix?         | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | see https://github.com/composer/composer/issues/9308
| Documentation   | n/a
| License         | MIT

#### What's in this PR?

This package provides the client implementation. But it does not provide the interfaces from psr/http-client

#### Why?

Which problem does the PR fix? (remove this section if you linked an issue above)

#### Example Usage

``` php
// If you added new features, show examples of how to use them here
// (remove this section if not a new feature)

$foo = new Foo();

// Now we can do
$foo->doSomething();
```

#### Checklist

- [ ] Updated CHANGELOG.md to describe BC breaks / deprecations | new feature | bugfix
- [ ] Documentation pull request created (if not simply a bugfix)

#### To Do

- [ ] If the PR is not complete but you want to discuss the approach, list what remains to be done here

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -22,8 +22,7 @@
         "php-http/client-common": "^2.0"
     },
     "provide": {
-        "php-http/client-implementation": "1.0",
-        "psr/http-client": "1.0"
```
Comment (andrew-demb) [inline composer.json:26]: Should we replace it in favor `psr/http-client-implementation`?

Comment (stof) [reply composer.json:26]: I don't understand your comment. this package does not provide and does not replace `psr/http-client`, which is the package providing the PSR ClientInterface. It even *requires* `psr/http-client` (which is probably why things work in Composer 1).

And my Pr has not change the fact that this package provides `php-http/client-implementation` (the line is edited only to remove the comma, as JSON does not support trailing commas in objects)

Comment (andrew-demb) [reply composer.json:26]: I mean, that this package should provide implementations for 2 packages:
- `psr/http-client-implementation` (https://packagist.org/?query=psr%2Fhttp-client-implementation)
- `php-http/client-implementation` (https://packagist.org/?query=php-http%2Fclient-implementation)

Comment (stof) [reply composer.json:26]: hmm, indeed, I missed that the existing line was for `php-http/client` and not `psr/http-client`

**Comment by stof** [comment]
This is now fixed

**Review by Seldaek — APPROVED** [review]

**Review by Nyholm — APPROVED** [review]
Thank you

**Pull request merged**